autoload/vim_dadbod_completion.vim

```diff
@@ -182,10 +182,10 @@ endfunction
 function! s:generate_query(db, query_key, ...) abort
   let base_query = db#adapter#dispatch(a:db, 'interactive')
   let Query = s:cache[a:db].scheme[a:query_key]
-  if a:0 > 0
+  if type(Query) == v:t_func
     let Query = Query(a:1)
   endif
```

Comment:
`Query` is used as a function, but `Query` isn't ready until another `#job#run()` has complete. `Query` would sometimes be a string and this condition would also pass, so I adjusted the condition to match what the enclosing statements require.